Ennemy development

Hendrik-Kruger opened the pull request:
Ribbug can jump floorMaw fulling implemented and code cleanup

Hendrik-Kruger committed:
FloorMaw attack added ribbug code cleanup

moved the player chase to its own function

Hendrik-Kruger committed:
Worked on ribbug jumping

Hendrik-Kruger committed:
Merge branch 'main' into Ennemy_development

Hendrik-Kruger committed:
updated from main  and also fixed the improved plager chase function for ribbug

It works as intended now again and code has been cleaned up a bit
also fixed floormaw

Hendrik-Kruger committed:
Worked on ribbug movement improvement and jumping also implemented new hitbox for player attack animation

added an area2D for attack area also made it so that ribbug can jump

Hendrik-Kruger committed:
worked on floormaw fixed the spanw area aslo worked on ribbug jumping

Hendrik-Kruger committed:
Merge branch 'main' into Ennemy_development

Hendrik-Kruger committed:
FloorMaw attacks player and player recieves damage from that

modified the attack area to fit the new animation. added check for if floormaw eneters recieve damage area. cleaned ribbug code

Hendrik-Kruger committed:
Merge branch 'main' into Ennemy_development

SkyePooley commented on Godot Project/Scenes/Entities/Player/attack_hitbox.gd:
Does this empty script have a purpose?

SkyePooley committed:
Removed redundant PowerupOff nodes.

PowerupOff nodes have been deprecated in favour of powerups stored in
the Abilities global node.
These were probably added back in by a merge.

SkyePooley reviewed (approved):
Looks good! Ribbug jump and floor maw attack work well.
There is an issue with the floor maw where it will start out extended and then retract when the player gets close.
